src/Authentication/ClientCredentials.php

```diff
@@ -198,6 +199,14 @@ protected function getHttpClient()
      */
     protected function handleRequestException(RequestException $exception)
     {
+        if (empty($exception->getResponse())) {
+            throw new AuthenticationException(
+                'The authentication server could not be reached, please check if the uri is correct',
+                404,
```

Comment:
When I get a `404`, I expect that I accidentally mistyped the url. Especially with this error message. While this could also mean that our auth server is offline right? Should we consider using a `500`?

Reply:
The chance is much greater that you miss typed the url, than that the server is so completely down that not even an error response is generated.